### DISCUSSION TITLE
[C128] How to move MAIN from $1C0D to $2000 and still LOAD as PRG

### DISCUSSION THREAD

**Discussion by RetrofanC** [cc65/cc65#2131] [Q&A]
Hi Community, 

For my project I am using 8k of common ram on the C128. 
Routines in all four memory banks working seamless together like a charm. 

However, now because the MAIN program in RAM0 always starts at $1C0D I am loosing valuable bytes of common ram. 
As the range from $1C0D to $2000 cannot be used for common anymore so to say.
_(I need to make this unavailable for common ram code)_ 

**So the question is;** 

What do I have to do to and in which files with which code/configurations.

To make it possible to have a MAIN starting at $2000. 
(So at least changing the c128.cfg of course and the LOADADDR two bytes) 

But still have it compiled correct and can be loaded from C128 command prompt with LOAD or RUN commands. 

Thanks!

**Comment by mrdudz** [comment]
there are generally two approaches to this:
a) remove the default BASIC starter, and produce a binary where everything is located where you want it (eg your MAIN at $2000). And then use a packer/cruncher on the resulting file, which in turn makes it RUNable again.
b) use a custom linker config, which will define different LOAD and RUN addresses for your MAIN block, and custom crt0 (startup) file which will then copy your MAIN block from its LOAD to its RUN address and jump to it.

a) is a bit easier and more straightforward, and you can test the unpacked binary (by starting with SYS) before packing, so its easier to get right
b) is a bit more involved and more error prone, with the advantage that you dont need an extra build step (the packer)
